tests/unittest/inputs/test_prefix_token_cache.py

```
@@ -0,0 +1,176 @@
+def test_interleaved_unrelated_prompts_are_correct():
+    tokenizer, cache = MergeTokenizer(), _cache()
+    unrelated = "a wholly different opening that shares no prefix at all. " * 3
+    for prompt in _turns(6):
+        assert cache.encode(tokenizer, prompt) == tokenizer(prompt)["input_ids"]
+        assert cache.encode(tokenizer, unrelated) == tokenizer(unrelated)["input_ids"]
+
+
+def test_resync_failure_falls_back_to_full_tokenization():
+    tokenizer, cache = UnsyncableTokenizer(), _cache()
+    results = [cache.encode(tokenizer, p) for p in _turns()]
+    for prompt, ids in zip(_turns(), results):
```

Ruff (0.16.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)